docs/codebase-best-practices.md

```diff
@@ -0,0 +1,25 @@
+# Codebase Best Practices
+
+## General JavaScript
+
+## Specific TypeScript
+
+### Migrating a JavaScript File to TypeScript
+
+#### Retaining Git File History
+
+Sometimes changing the file from `<filename>.js` -> `<filename>.tsx?` causes the original file to be deleted, and a new one created, and other times the filename just changes - in terms of Git. Ideally, we want the file history to be preserved.
+
+The best bet at achieving this is by:
+
+1. Rename the file
```

Comment:
If you rename it with `git mv` I think it preserves the history better?

Reply:
What happens with auto import renaming?

Reply:
That's... a good question.

It probably breaks, tbh, but I'd have to test to be sure.